tests/utils/test_performance_optimizer.py

```
@@ -177,0 +1,36 @@
+        result = self.optimizer.optimize_batch(test_data)
+        self.assertEqual(len(result), len(test_data))
+        self.assertTrue(all(isinstance(item, dict) for item in result))
+
+    @patch('src.utils.performance_optimizer.monitor_resources')
+    def test_resource_monitoring(self, mock_monitor):
+        mock_monitor.return_value = {'cpu': 50, 'memory': 70}
+        metrics = self.optimizer.check_resource_usage()
+        self.assertIn('cpu', metrics)
+        self.assertIn('memory', metrics)
+
+    def test_performance_metrics_collection(self):
```

Ruff (0.8.2)

Missing return type annotation for public function `test_performance_metrics_collection`

Add return type annotation: `None`

(ANN201)